src/rgw/rgw_op.cc

```diff
@@ -2682,6 +2682,37 @@ void RGWDeleteBucket::execute()
     }
   }
 
+  vector<rgw_bucket_dir_entry> objs;
+  string prefix = "", marker = "", delimiter = "";
+  map<string, bool> common_prefixes;
+  bool is_truncated;
+  static MultipartMetaFilter mp_filter;
+  int max_uploads = 1000;
+
+  if (s->prot_flags & RGW_REST_SWIFT) {
+    string path_args;
+    path_args = s->info.args.get("path");
+    if (!path_args.empty()) {
+      if (!delimiter.empty() || !prefix.empty()) {
+        op_ret = -EINVAL;
+        return;
+      }
+      prefix = path_args;
+      delimiter="/";
+    }
+  }
+
+  op_ret = list_bucket_multiparts(store, s->bucket_info, prefix, marker, delimiter, &mp_filter,
+				  max_uploads, &objs, &common_prefixes, &is_truncated);
+  if (op_ret < 0) {
+    return;
+  }
+  if (!objs.empty()) {
+    ldout(s->cct, 0) << "ERROR: bucket " << s->bucket_name << " not empty, incomplete multiparts found" << dendl;
+    op_ret = -ENOTEMPTY;
```

Comment:
@cbodley Right, this is not compatible with current AWS behaviour. I even tried deleting a bucket in S3 with an incomplete multipart upload and it was successful. What I am not sure is whether, S3 would clean this up in background or charge me for the leaked objects, as mentioned [here](http://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html) 
Do you recommend aborting the incomplete uplads here without throwing any error?